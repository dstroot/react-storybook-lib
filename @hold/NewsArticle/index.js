import React from 'react';

const NewsArticle = () => {
  return (
    <div className="container">
      <div className="row">
        <div className="col">
          <h5 className="text-uppercase text-primary mb-0 mt-5">
            Press Release
          </h5>
          <p>December 20, 2018</p>
          <h1 className="display-6 mt-5 mb-0">
            Pacific Life Foundation Donates $250,000 to the American Red Cross
            for Victims of the California Wildfires
          </h1>
          <hr className="m-4" />
          <p>
            NEWPORT BEACH, CA – The Pacific Life Foundation announced today a
            $250,000 donation to the American Red Cross in support of relief
            efforts for those impacted by the California wildfires. The
            Foundation also announced plans to match employee donations.
          </p>
          <p>
            To help people affected by the California wildfires, visit
            redcross.org, call 1- 800-RED CROSS or text the word CAWILDFIRES to
            90999 to make a $10 donation. Donations enable the Red Cross to
            prepare for, respond to and help people recover from these
            disasters.
          </p>
          <h5>About Pacific Life and the Pacific Life Foundation</h5>
          <p>
            The Pacific Life Foundation was established in 1984. Together with
            Pacific Life, the Foundation has contributed more than $100 million
            to community and national nonprofit organizations. Grants are made
            to organizations that address a broad spectrum of social needs.
          </p>
          <p>
            Celebrating its 150th anniversary in 2018, Pacific Life provides a
            wide range of insurance products, annuities, and mutual funds, and
            offers a variety of investment products and services to individuals,
            businesses, and pension plans. Pacific Life counts more than half of
            the 100 largest U.S. companies as its clients. For additional
            company information, including current financial strength ratings,
            visit{' '}
            <a
              className="card-link"
              href="http://www.pacificlife.com/"
              target="_top"
            >
              www.PacificLife.com
            </a>
            . To explore the heritage of Pacific Life and learn more about our
            history, visit{' '}
            <a className="link" href="http://www.pl150years.com" target="_top">
              www.pl150years.com
            </a>
            .
          </p>
          <p>
            Pacific Life refers to Pacific Life Insurance Company and its
            affiliates, including Pacific Life & Annuity Company. Client count
            as of June 2018 is compiled by Pacific Life using the 2018 FORTUNE
            500® list.
          </p>
          <hr className="m-4" />

          <div className="row">
            <div className="col">
              <h3>Press Contacts</h3>
              <address className="">
                <strong>Media Contact</strong>
                <br />
                <a
                  className="card-link"
                  href="mailto:media@pacificlife.com?Subject=Media%20Inquiry"
                  target="_top"
                >
                  media@pacificlife.com
                </a>
                <br />
                <a className="card-link" href="tel:+19492194575">
                  (949) 219-4575
                </a>
              </address>
            </div>
            <div className="col">
              <img
                // className="mr-3 d-none d-sm-block"
                src="/media/logo.png"
                alt="Generic placeholder"
                width="175px"
                height="115px"
              />
            </div>
          </div>

          <hr className="m-4" />
          <a className="btn btn-large btn-secondary mb-5" href="/news">
            Back to Latest Articles
          </a>
        </div>
      </div>
    </div>
  );
};

export default NewsArticle;
