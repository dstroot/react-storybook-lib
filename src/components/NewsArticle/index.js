import React from "react";
import Markdown from "react-markdown";
import logo from "./media/logo.png";
import { Link } from "@reach/router";

const NewsArticle = ({ article }) => {
  const { datetime, draft, cutline, headline, story } = article;
  return (
    <div className="container">
      <div className="row">
        <div className="col">
          <h5 className="text-uppercase text-primary mb-0 mt-5">
            Press Release{" "}
            {draft ? <span className="text-secondary"> [DRAFT]</span> : ""}
          </h5>
          <p>{datetime}</p>
          <h1 className="display-6 mt-5">{headline}</h1>
          <p className="lead mt-4">{cutline}</p>
          <hr className="m-4" />
          <Markdown source={story} />
          <hr className="m-4" />

          <div className="row">
            <div className="col">
              <h3>Press Contacts</h3>
              <address className="">
                <strong>Media Contact</strong>
                <br />
                <a
                  className="card-link"
                  href="mailto:media@pacificlife.com?Subject=Media%20Inquiry"
                  target="_top"
                >
                  media@pacificlife.com
                </a>
                <br />
                <a className="card-link" href="tel:+19492194575">
                  (949) 219-4575
                </a>
              </address>
            </div>
            <div className="col">
              <img
                // className="mr-3 d-none d-sm-block"
                src={logo}
                alt="logo"
                width="175px"
                height="115px"
              />
            </div>
          </div>

          <hr className="m-4" />
          <Link to="/news" className="btn btn-large btn-secondary mb-5">
            Back to Articles
          </Link>
        </div>
      </div>
    </div>
  );
};

export default NewsArticle;
