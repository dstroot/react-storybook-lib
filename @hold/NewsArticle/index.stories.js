import React from 'react';
import { storiesOf } from '@storybook/react';

// components
import NewsArticle from '../NewsArticle';

// NewsArticle section
storiesOf('NewsArticle', module).add('show NewsArticle', () => <NewsArticle />);
