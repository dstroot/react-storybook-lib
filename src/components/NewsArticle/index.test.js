import React from 'react';
import renderer from 'react-test-renderer';

import NewsArticle from '../NewsArticle';

describe('NewsArticle', () => {
  it('it should render', () => {
    const component = renderer.create(<NewsArticle />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
