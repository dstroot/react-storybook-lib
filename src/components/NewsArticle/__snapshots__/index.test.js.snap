// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewsArticle it should render 1`] = `
<div
  className="container"
>
  <div
    className="row"
  >
    <div
      className="col"
    >
      <h5
        className="text-uppercase text-primary mb-0 mt-5"
      >
        Press Release
         
        
      </h5>
      <p>
        2019-01-08
      </p>
      <h1
        className="display-6 mt-5"
      >
        Pacific Life Foundation Donates $250,000 to the American Red Cross for Victims of the California Wildfires
      </h1>
      <p
        className="lead mt-4"
      >
        Pacific Life Foundation Donates $250,000 to the American Red Cross for Victims of the California Wildfires
      </p>
      <hr
        className="m-4"
      />
      <p>
        NEWPORT BEACH, Calif. – Jan. 8, 2019 – Pacific Life today announced that its Life Insurance and Retirement Solutions Divisions have both been honored with 2018 DALBAR customer service awards. The Life Insurance Division was awarded with the Insurance Service Award which recognizes outstanding telephone support to policyholders. These interactions are reviewed throughout the year, covering all aspects of the customer experience. To qualify for the award, companies must exceed quality thresholds in all criteria.Pacific Life's Retirement Solutions Division earned the Financial Intermediary Service Award for the 15th consecutive year and ranked #1 among annuity companies for the sixth consecutive year. The award reflects outstanding quality of telephone service and support delivered post-sale to financial professionals."At Pacific Life, we value the relationships we share with policyholders and financial professionals, which don’t end when a product is sold," said Pacific Life Executive Vice President and Chief Operating Officer Adrian Griggs. "We're proud to be recognized for exceptional service and strive to ensure enduring customer support."
        <strong>
          About Pacific Life
        </strong>
        For more than 150 years, Pacific Life has helped millions of individuals and families with their financial needs through a wide range of life insurance products, annuities, and mutual funds, and offers a variety of investment products and services to individuals, businesses, and pension plans. Whether your goal is to protect loved ones or grow your assets for retirement, Pacific Life offers innovative products and services that provide value and financial security for current and future generations. Pacific Life counts more than half of the 100 largest U.S. companies as its clients. For additional company information, including current financial strength ratings, visit www.PacificLife.com. Pacific Life refers to Pacific Life Insurance Company and its affiliates, including Pacific Life & Annuity Company. Client count as of June 2018 is compiled by Pacific Life using the 2018 FORTUNE 500® list.
        <strong>
          About DALBAR
        </strong>
        DALBAR, Inc. is the financial community’s leading independent expert for evaluating, auditing and rating business practices, customer performance, product quality and service. Launched in 1976, DALBAR has earned recognition for consistent and unbiased evaluations of investment companies, registered investment advisers, insurance companies, broker/dealers, retirement plan providers and financial professionals. DALBAR awards are recognized as marks of a superior standard of care in the financial community.
      </p>
      <hr
        className="m-4"
      />
      <div
        className="row"
      >
        <div
          className="col"
        >
          <h3>
            Press Contacts
          </h3>
          <address
            className=""
          >
            <strong>
              Media Contact
            </strong>
            <br />
            <a
              className="card-link"
              href="mailto:media@pacificlife.com?Subject=Media%20Inquiry"
              target="_top"
            >
              media@pacificlife.com
            </a>
            <br />
            <a
              className="card-link"
              href="tel:+19492194575"
            >
              (949) 219-4575
            </a>
          </address>
        </div>
        <div
          className="col"
        >
          <img
            alt="logo"
            height="115px"
            src="logo.png"
            width="175px"
          />
        </div>
      </div>
      <hr
        className="m-4"
      />
      <a
        className="btn btn-large btn-secondary mb-5"
        href="/news"
        onClick={[Function]}
      >
        Back to Articles
      </a>
    </div>
  </div>
</div>
`;
